py3status/modules/spotify.py

```diff
@@ -102,6 +113,24 @@ def _get_text(self):
                 self.format_down,
                 self.py3.COLOR_OFFLINE or self.py3.COLOR_BAD)
 
+    def _compile_re(self, expression):
+        """
+        Compile given regular expression for current sanitize words
+        """
+        expression = expression.replace('META_WORDS_HERE', self.sanitize_words)
+        return re.compile(expression, re.IGNORECASE)
+
+    def _sanitize_title(self, title):
+        """
+        Remove redunant meta data from title and return it
+        """
+        inside_brackets = self._compile_re(r"([\(\[][^)\]]*?(META_WORDS_HERE)[^)\]]*?[\)\]])")
+        after_delimiter = self._compile_re(r"([\-,;/])([^\-,;/])*(META_WORDS_HERE).*")
```

Comment:
Good point with the post_config_hook, will fix it.

Matching the same kind of braces in regexps may be too tricky for me (I'm a beginner when it comes to regular expressions) and I don't think there would be much benefit in that, so as you've said, that should not be a problem.

After processing title with regexes in _sanitize_title I do title.strip(),1 so I think that actually there should not be any trailing whitespace ;)